FreeRTOS/FreeRTOS
# Update header_checker.py to support python and assembly files.

Description
-----------
Update header_checker.py to support python and assembly files (with slightly modified headers)
Allow specifying an interpreter with a "#!" directive at the beginning of files (useful for python files)
Add support for checking and verifying SPDX license identifiers.

Test Steps
-----------
Step through the output of kernel_checker.py and core_checker.py and compare with the previous output. Validate that no files which are necessary have been missed.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
